modules/400-descheduler/templates/deployment.yaml

```diff
@@ -3,73 +3,75 @@ cpu: 25m
 memory: 50Mi
 {{- end }}
 
-{{- if (.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+{{- range $d := .Values.descheduler.internal.deschedulers }}
+  {{- $deschedulerSpec := $d.spec }}
+  {{- if ($.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
 ---
 apiVersion: autoscaling.k8s.io/v1
 kind: VerticalPodAutoscaler
 metadata:
-  name: descheduler
+  name: descheduler-{{ $d.metadata.name }}
   namespace: d8-descheduler
-  {{- include "helm_lib_module_labels" (list . (dict "app" "descheduler")) | nindent 2 }}
+  {{- include "helm_lib_module_labels" (list $ (dict "app" "descheduler")) | nindent 2 }}
 spec:
   targetRef:
     apiVersion: "apps/v1"
     kind: Deployment
-    name: descheduler
+    name: descheduler-{{ $d.metadata.name }}
   updatePolicy:
     updateMode: "Auto"
   resourcePolicy:
     containerPolicies:
     - containerName: "descheduler"
       minAllowed:
-        {{- include "descheduler_resources" . | nindent 8 }}
+        {{- include "descheduler_resources" $ | nindent 8 }}
       maxAllowed:
         cpu: 50m
         memory: 100Mi
-{{- end }}
+  {{- end }}
 ---
 apiVersion: policy/v1beta1
 kind: PodDisruptionBudget
 metadata:
-  name: descheduler
+  name: descheduler-{{ $d.metadata.name }}
   namespace: d8-descheduler
-  {{- include "helm_lib_module_labels" (list . (dict "app" "descheduler")) | nindent 2 }}
+  {{- include "helm_lib_module_labels" (list $ (dict "app" "descheduler")) | nindent 2 }}
 spec:
   maxUnavailable: 1
   selector:
     matchLabels:
-      app: descheduler
+      app: descheduler-{{ $d.metadata.name }}
 ---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: descheduler
+  name: descheduler-{{ $d.metadata.name }}
   namespace: d8-descheduler
-  {{- include "helm_lib_module_labels" (list .) | nindent 2 }}
+  {{- include "helm_lib_module_labels" (list $ (dict "app" "descheduler")) | nindent 2 }}
 spec:
-  replicas: {{ .Values.descheduler.internal.replicas }}
+  replicas: 1
   revisionHistoryLimit: 2
   selector:
     matchLabels:
-      app: descheduler
+      app: descheduler-{{ $d.metadata.name }}
   template:
     metadata:
       labels:
-        app: descheduler
+        app: descheduler-{{ $d.metadata.name }}
       annotations:
-        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        checksum/config: {{ $deschedulerSpec | toJson | sha256sum }}
```

Comment:
NIT: it is possible to avoid json marshalling here and calculate the checksum in the hook using, e.g., hashstruct.

This is up to you.

Reply:
Still prefer the Helm one. Due to its ubiquitousness.